### PR TITLE
CI: Fix stubtest workflow; make paths for other workflows more specific

### PR DESCRIPTION
**Pull request by AlexWaygood**
Reverts python/typeshed#7895. The new workflow doesn't seem to be being triggered: see #7896

### CODE REVIEW

File: .github/workflows/stubtest_stdlib.yml [hunk 1]
```diff
@@ -1,43 +0,0 @@
-name: Run stubtest on the stdlib
-
-on:
-  workflow_dispatch:
-  push:
-    branches:
-      - main
-      - master
-    pull_request:
```
Comment (JelleZijlstra) [inline .github/workflows/stubtest_stdlib.yml:9]: Is this overindented? In tests.yml it's at the same level as `push:`.

Comment (srittau) [reply .github/workflows/stubtest_stdlib.yml:9]: Good spot! I believe this is the reason.

Comment (AlexWaygood) [reply .github/workflows/stubtest_stdlib.yml:9]: Oh wow. 🤦 Thanks!

Comment (AlexWaygood) [reply .github/workflows/stubtest_stdlib.yml:9]: Yup, that was it!

**Review by JelleZijlstra — APPROVED** [review]

File: .github/workflows/stubtest_stdlib.yml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-      paths:
-        - 'stdlib/**'
-        - '.github/workflows/**'
-        - 'tests/**'
+  pull_request:
+    paths:
+      - 'stdlib/**'
+      - '.github/workflows/stubtest_stdlib.yml'
+      - 'tests/stubtest_stdlib.py'
```
Comment (JelleZijlstra) [inline .github/workflows/stubtest_stdlib.yml:13]: I'd rather just keep this as `tests/**`, in case we update the test file to use colors.py or something.

Comment (AlexWaygood) [reply .github/workflows/stubtest_stdlib.yml:13]: Fair point, I wasn't 100% sure about that one.

Comment (AlexWaygood) [reply .github/workflows/stubtest_stdlib.yml:13]:
```suggestion
      - 'tests/**'
```

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**